Plutarch/Bool.hs

```diff
@@ -84,7 +80,11 @@ infix 4 #==
 
 class PEq t => POrd t where
   (#<=) :: Term s t -> Term s t -> Term s PBool
```

Comment:
That would destroy the point?

Reply:
or do I misunderstand how it works? Can you make it err if you either implement all or none?

Reply:
The only fix I can think of besides fixing `MINIMAL` is avoiding default implementations and doing this:
```haskell
class DeriveOrd a
instance {-# OVERLAPPABLE #-} DeriveOrd a => Ord a where
  ...
```
Then you could do `deriving anyclass DeriveOrd`.
This however doesn't satisfy me because:
- Haskell doesn't warn on incoherent usage of this overlappable instance.
- Even if it did, you can still implement both `DeriveOrd` *and* `Ord` manually!

What we really want is
```haskell
instance DeriveOrd a => Ord a where
  ...
```
which should disallow manually deriving `Ord A` if `DeriveOrd A`.
You could implement this by taking constraints into account when resolving instances possibly.
If you have `[W] Ord A` and you have both the instances `DeriveOrd A => Ord A` and `Ord A`, GHC could discard the first one by noting that the constraint isn't fulfilled. This calls for another GHC proposal.

Reply:
Got a great answer by tomjaguarpaw, but it means we have to have only one method per class... huge PITA

Reply:
and also breaking change

Reply:
Perhaps the OVERLAPPABLE approach is better?

Reply:
I honestly think it's fine - no need to make it more complicated.

Reply:
The issue is this can lead to subtle security bugs. Implementing just `#<=` while `#<` gives a completely incompatible resultl.

Reply:
hmm

Reply:
this is why we have typeclass laws, yes?

Reply:
Typeclass laws don't warn you about anything...